zerver/actions/message_edit.py

```diff
@@ -914,6 +915,12 @@ def check_update_message(
     if not user_profile.realm.allow_message_editing:
         raise JsonableError(_("Your organization has turned off message editing"))
 
+    if (
+        message.recipient.type == Recipient.STREAM
+        and not Stream.objects.filter(id=message.recipient.type_id, deactivated=False).exists()
+    ):
+        raise JsonableError(_("Stream does not exist"))
```

Comment:
I think logically, we want to fetch the stream containing the message (if any), ideally in a common code path so we don't fetch it N times, and then just check if it's deactivated.